src/components/QuickOpenModal.js

```diff
@@ -95,7 +95,9 @@ export class QuickOpenModal extends Component<Props, State> {
       const results = this.props.sources;
       return this.setState({ results });
     }
-
+    if (this.isGotoQuery()) {
```

Comment:
You could avoid the setState entirely by changing line 91 to this: 

```js
    if (!this.isGotoQuery() && (nowEnabled || queryChanged)) {
```
and line 248 to:

```js
    if ((this.isSymbolSearch() && noSource) || this.isGotoQuery()) {
```